packages/react/toast/src/Toast.tsx

```diff
@@ -651,6 +651,60 @@ ToastImpl.propTypes = {
   },
 };
 
+/* -----------------------------------------------------------------------------------------------*/
+
+interface ToastAnnounceProps
+  extends React.ComponentPropsWithoutRef<'div'>,
+    ScopedProps<{ children?: ToastImplProps['children'] }> {}
+
+const ToastAnnounce: React.FC<ToastAnnounceProps> = (props: ScopedProps<ToastAnnounceProps>) => {
+  const { __scopeToast, children, ...announceProps } = props;
+  const context = useToastProviderContext(TOAST_NAME, __scopeToast);
+  const [renderChildren, setRenderChildren] = React.useState(false);
+  const [isAnnounced, setIsAnnounced] = React.useState(false);
+  const [fragment, setFragment] = React.useState<DocumentFragment>();
+  const [rootFragmentNode, setRootFragmentNode] = React.useState<HTMLDivElement | null>(null);
+
+  // We portal children into a document fragment so that we can extract the bare text nodes
+  // before rendering to the DOM. This avoids issues with duplicate `children`
+  // and animation libraries when composing via `asChild`.
+  const announceTextContent = React.useMemo(() => {
+    return rootFragmentNode ? getAnnounceTextContent(rootFragmentNode) : null;
+  }, [rootFragmentNode]);
+
+  // setting the fragment in `useLayoutEffect` as `DocumentFragment` doesn't exist on the server
+  useLayoutEffect(() => {
+    setFragment(new DocumentFragment());
+  }, []);
+
+  // render children in the next frame to ensure toast is announced in NVDA
+  useNextFrame(() => setRenderChildren(true));
+
+  React.useEffect(() => {
+    const timer = window.setTimeout(() => setIsAnnounced(true), 1000);
```

Comment:
What is this timer for? Seeing hardcoded times like this makes my senses tingle 😀